pkg/vault/contracts/Vault.sol

```diff
@@ -664,14 +668,24 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
         amountsInRaw = new uint256[](locals.numTokens);
 
         for (uint256 i = 0; i < locals.numTokens; ++i) {
+            uint256 amountInRaw;
+
             // 1) Calculate raw amount in.
-            // amountsInRaw are amounts actually entering the Pool, so we round up.
-            // Do not mutate in place yet, as we need them scaled for the `onAfterAddLiquidity` hook
-            uint256 amountInRaw = amountsInScaled18[i].toRawUndoRateRoundUp(
-                poolData.decimalScalingFactors[i],
-                poolData.tokenRates[i]
-            );
-            amountsInRaw[i] = amountInRaw;
+            {
+                uint256 amountInScaled18 = amountsInScaled18[i];
+                if (amountInScaled18 > 0) {
```

Comment:
Why do we consider 0 to be a valid trade amount? Removing this `if` would save some bytecode (maybe) and reduce gas for the majority of operations, or not?

Reply:
This check is in `_addLiquidity`. We need to support 0 amounts because unbalanced joins will often times contain 0 values.

Example: I join the USDC/DAI/USDT pool UNBALANCED with only USDC & DAI